Makefile

```diff
@@ -1,220 +1,293 @@
+##############################################################################
+# Linting, style-checking, license header checks, etc.
+##############################################################################
 
-.PHONY: import-dashboards
-import-dashboards:
-	echo "APM loads dashboards via Kibana, not the APM Server"
+GOLINT_TARGETS?=$(shell go list ./...)
+GOLINT_UPSTREAM?=origin/master
+REVIEWDOG_FLAGS?=-conf=_beats/reviewdog.yml -f=golint -diff="git diff $(GOLINT_UPSTREAM)"
+GOLINT_COMMAND=$(shell $(GOLINT) ${GOLINT_TARGETS} | grep -v "should have comment" | $(REVIEWDOG) $(REVIEWDOG_FLAGS))
+
+.PHONY: golint
+golint: $(GOLINT) $(REVIEWDOG)
+	@test -z "$(GOLINT_COMMAND)" || (echo "$(GOLINT_COMMAND)" && exit 1)
+
+.PHONY: staticcheck
+staticcheck: $(STATICCHECK)
+	$(STATICCHECK) github.com/elastic/apm-server/...
 
 .PHONY: check-changelogs
-check-changelogs: python-env ## @testing Checks the changelogs for certain branches.
-	@python script/check_changelogs.py
+check-changelogs: $(PYTHON)
+	$(PYTHON) script/check_changelogs.py
 
-.PHONY: rm-empty-folders
-rm-empty-folders:
-	find vendor/ -type d -empty -delete
+.PHONY: check-headers
+check-headers: $(GOLICENSER)
+ifndef CHECK_HEADERS_DISABLED
+	@$(GOLICENSER) -d -exclude build -exclude x-pack
+	@$(GOLICENSER) -d -exclude build -license Elastic x-pack
+endif
 
-.PHONY: run-system-test
-run-system-test: python-env
-	INTEGRATION_TESTS=1 TZ=UTC \
-	ES_USER=$(ES_USER) ES_PASS=$(ES_PASS) KIBANA_USER=$(BEAT_KIBANA_USER) KIBANA_PASS=$(BEAT_KIBANA_PASS) \
-	$(PYTHON_ENV)/bin/nosetests --with-timer -x -v $(SYSTEM_TEST_TARGET)
+# TODO(axw) once we move to modules, start using "mage fmt" instead.
+.PHONY: gofmt autopep8
+fmt: gofmt autopep8
+gofmt: $(GOIMPORTS) add-headers
+	@echo "fmt - goimports: Formatting Go code"
+	@$(GOIMPORTS) -local github.com/elastic -l -w \
+		$(shell find . -type f -name '*.go' -not -path "*/vendor/*" 2>/dev/null)
+autopep8: $(MAGE)
+	@$(MAGE) pythonAutopep8
 
-# docker-compose.override.yml holds overrides for docker-compose.yml.
-#
-# Create this to ensure the UID used inside docker-compose is the same
-# as the current user on the host, so files are created with the same
-# privileges.
-#
-# Note that this target is intentionally non-.PHONY, so that users can
-# modify the resulting file without it being overwritten. To recreate
-# the file, remove it.
-docker-compose.override.yml:
-	printf "version: '2.3'\nservices:\n beat:\n  build:\n   args: [UID=%d]" $(shell id -u) > $@
-system-tests-environment: docker-compose.override.yml
-build-image: docker-compose.override.yml
+##############################################################################
+# Rules for creating and installing build tools.
+##############################################################################
+
+# $GOBIN must be set to use "go get" below. Once we move to modules we
+# can just use "go build" and it'll resolve all dependencies using modules.
+export GOBIN=$(CURDIR)/build
```

Comment:
I changed this to `export GOBIN=$(addprefix $(CURDIR)/,$(GOOSBUILD))` (though I'm not sure that's the best change) to resolve:

```
$ make
go get ./vendor/github.com/magefile/mage
make: ./build/darwin/mage: No such file or directory
make: *** [include/fields.go] Error 1
```

Reply:
I missed pushing a commit - it's `export GOBIN=$(abspath $(GOOSBUILD))` now.